lib/tokenize.cpp

```diff
@@ -3258,19 +3258,20 @@ void Tokenizer::simplifyCaseRange()
                 }
             }
         } else if (Token::Match(tok, "case %char% ... %char% :")) {
-            const char start = tok->strAt(1)[1];
-            const char end = tok->strAt(3)[1];
-            if (start < end) {
+            const MathLib::bigint start = MathLib::toLongNumber(tok->strAt(1));
+            const MathLib::bigint end = MathLib::toLongNumber(tok->strAt(3));
+            if (start < end && start >= std::numeric_limits<char>::min() && end <= std::numeric_limits<char>::max()) {
```

Comment:
`std::numeric_limits<char>::min()` will not return the value according to the `--platform` option..  Use `Platform` values to get type limits. like `mSettings->..`

Reply:
`characterLiteralToLL` currently assumes same representation and signedness of `char` as on the host, so if I use the platform option here, it will not match up with the interpretation used for `start` and `end`.

In the following lines `i` is also implicitly interpreted as character according to the host's `char`. I would at least need to switch between `signed char`/`unsigned char` there as well.

Maybe I should temporarily use conservative limits of `0x00`and `0x7f` here until the signedness is handled correctly by `characterLiteralToLL`?

As I mentioned in the PR, I think most of these complications can be avoided by just merging the numeric and character range cases together. Would that be an acceptable solution?

Reply:
> Maybe I should temporarily use conservative limits of 0x00and 0x7f here until the signedness is handled correctly by characterLiteralToLL?

that is acceptable for me.

> As I mentioned in the PR, I think most of these complications can be avoided by just merging the numeric and character range cases together. Would that be an acceptable solution?

yes spontanously sounds good. I don't know any good reason to separate it.

Reply:
I have merged the handling in the new commit. This check is now obsolete.